XExtensionItem/XExtensionItem.m

```diff
@@ -193,7 +193,12 @@ - (id)activityViewController:(UIActivityViewController *)activityViewController
         item.attachments = ({
             id activityItem = [self activityItemForActivityType:activityType];
             
-            NSItemProvider *mainAttachment = [[NSItemProvider alloc] initWithItem:activityItem typeIdentifier:self.typeIdentifier];
+            NSString *typeIdentifier = self.typeIdentifier;
+            if (![[activityItem class] isSubclassOfClass:[self.placeholderItem class]]) {
```

Comment:
Yes - the idea is that we want to check if the class has changed between the actual activity item and the placeholder. In order to do that, we call `isSubclassOfClass:` to check if the new class is the same as or a subclass of the old one (for example, if you pass in an animated image, which is of class _UIAnimatedImage, a subclass of UIImage).

It's definitely not foolproof, but it should work in most cases and even if this check gets it wrong, it's more likely to be a false negative than a false positive, so it should work out properly either way.

Though now that I'm thinking about it, it may have a false positive if you pass in a file URL rather than a regular `NSURL`, since they're both just `NSURL` objects. I will have to address this case.